src/pathwayvis/components/map/map.component.ts

```diff
@@ -367,9 +367,13 @@ class MapComponentCtrl {
         this._mapOptions.updateMapData(response, this._mapOptions.getSelectedId());
         if (action.type.startsWith('reaction:objective')) {
           this._mapOptions.setObjectiveReaction(action.type.endsWith('undo') ? null : data.bigg_id);
-        } else if (data.bigg_id === this._mapOptions.getObjectiveReaction()) {
+        } else if (data.bigg_id === this._mapOptions.getObjectiveReaction()
+          && !action.type.startsWith('reaction:minimize')
```

Comment:
I think you're doing an exact match here since there's no do and undo action for minimize and maximize.
You could go like `!['reaction:minimize', 'reaction:maximize'].includes(action.type)`.

But I still think that a decomposition would make this code much more readable:

    const [target, actionType, direction] = action.type.split(':')
    ...
    !['maximize', 'minimize'].includes(actionType)